src/ecStackedScopes.ml

```diff
@@ -0,0 +1,79 @@
+(* -------------------------------------------------------------------- *)
```

Comment:
I factored out the stacked scopes structure to an external module (and removed the global reference).

I exported from `EcCommands` what is needed to write such a module.

While I was at it, I wrote it in a more idiomatic way (pattern match on lists instead of chaining calls to `List.hd` / `List.tl`). Please, backport this module to your development (I don't think this should go to the EC codebase) and remove this file (and its interface file) when you are done.

Reply:
I deleted the stacked scopes files and integrated them into the UC DSL development. Thanks for the refactoring of `ecCommands` that made this possible. We should have thought of that.

Regarding the structural equality test in `update_with_required`, I replaced it with a physical equality test, which is what I wanted. Given how loaded files work, I don't think the assertion will ever be evaluated, and viewed it as a sanity check. Please advise what you had in mind.